### PR TITLE
HW05 added tests for HW02, HW03

### PR DESCRIPTION
**Pull request by andreyruby**

### CODE REVIEW

File: HW02/andrey_bakanovsky/spec/array_refine_spec.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+require_relative '../array_refine'
+
+using ArrayRefine
+
+RSpec.describe ArrayRefine, Array do
+  subject(:test_array) { [11, 20, 93, 4, -5] }
+
+  describe '#my_map' do
+    context 'when calls for each element' do
+      it 'returns the original element to a new array' do
```
[the remark this answers is not in the document]
Comment (andreyruby) [reply HW02/andrey_bakanovsky/spec/array_refine_spec.rb]: Yes, my tests were obvious. I del part of them and added your recommendations tests. Its became more professional

File: HW02/andrey_bakanovsky/spec/array_refine_spec.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  end
+
+  describe '#my_select' do
+    context 'when the block\'s condition is true' do
+      it 'returns array of positive numbers' do
+        expect(test_array.my_select(&:positive?)).to match_array([11, 20, 93, 4])
+      end
+
+      it 'returns array of even numbers' do
+        expect(test_array.my_select(&:even?)).to match_array([20, 4])
```
[the remark this answers is not in the document]
Comment (andreyruby) [reply HW02/andrey_bakanovsky/spec/array_refine_spec.rb:44]: I added tests for the method: when array is empty, when calls without block, modify original array.
It improved the test my_select.

File: HW02/andrey_bakanovsky/spec/array_refine_spec.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      end
+
+      it 'returns empty array' do
+        expect(test_array.my_select { |x| x < -5 }).to be_empty
+      end
+    end
+  end
+
+  describe '#my_each' do
```
[the remark this answers is not in the document]
Comment (andreyruby) [reply HW02/andrey_bakanovsky/spec/array_refine_spec.rb:71]: Yes, I added the same tests.

[the remark this answers is not in the document]
Comment (andreyruby) [reply HW02/andrey_bakanovsky/spec/array_refine_spec.rb:71]: Thanks a lot

File: HW03/andrey_bakanovsky/spec/andrey_bakanovsky_hw_03_t_01_spec.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require_relative '../andrey_bakanovsky_hw_03_t_01'
+
+RSpec.describe 'hw_03_t_01' do
+  let(:log1) { '10.6.246.103 - - [23/Apr/2018:20:30:39 +0300] "POST /grid/2/messages HTTP/1.1" 200 48 0.0498' }
+  let(:log2) do
```
[the remark this answers is not in the document]
Comment (andreyruby) [reply HW03/andrey_bakanovsky/spec/andrey_bakanovsky_hw_03_t_01_spec.rb:7]: Yes, I defined multistring when used let.

File: HW03/andrey_bakanovsky/spec/andrey_bakanovsky_hw_03_t_01_spec.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    '2018-04-23 20:30:42: SSL error, peer: 10.6.246.101, peer cert: ,'\
+  ' #<Puma::MiniSSL::SSLError: System error: Undefined error: 0 - 0>'
+  end
+
+  describe 'contains_error?' do
+    it 'reaturns false' do
+      ask_contains_error = contains_error?(/error/i)
```
[the remark this answers is not in the document]
Comment (andreyruby) [reply HW03/andrey_bakanovsky/spec/andrey_bakanovsky_hw_03_t_01_spec.rb]: I'v found out with your help. I del the local variables. Thanks

File: HW03/andrey_bakanovsky/spec/andrey_bakanovsky_hw_03_t_03_spec.rb [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+require_relative '../andrey_bakanovsky_hw_03_t_03'
+
+RSpec.describe 'hw_03_t_03' do
+  let(:log_one_line_colling) { '2018-04-23 17:17:49.7 ubuntu-xenial[14319] Debug - Calling core with action: event' }
+  let(:log_one_line_without_colling) { '2018-04-23 17:17:49.8 ubuntu-xenial[14319] Debug - docker event processed' }
+  let(:output_datetime) { '2018-04-23 17:17:49.7' }
+  let(:log_without_datetime) { '--- WhenLogDoesNotContainTimeString' }
+
+  logs_calling = <<~LGSOUTPUT0
```
Comment (B0RlS) [inline HW03/andrey_bakanovsky/spec/andrey_bakanovsky_hw_03_t_03_spec.rb:11]: use let

File: HW02/andrey_bakanovsky/spec/array_refine_spec.rb [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+      it 'don\'t modify original array' do
+        expect(test_array.my_select { |x| x > 32 }.object_id) != test_array.object_id
+      end
+    end
+  end
+
+  describe '#my_each' do
+    context 'when calls for array' do
+      it 'returns a new array' do
+        new_array = test_array.my_each { |x| x }
```
Comment (AliakseiAndryianau) [inline HW02/andrey_bakanovsky/spec/array_refine_spec.rb:74]: there still local variables here a in the next tests. get rid of them please

File: HW03/andrey_bakanovsky/spec/andrey_bakanovsky_hw_03_t_03_spec.rb (hunk 6 above)
Comment (andreyruby) [reply HW03/andrey_bakanovsky/spec/andrey_bakanovsky_hw_03_t_03_spec.rb]: corrected. I used let

File: HW02/andrey_bakanovsky/spec/array_refine_spec.rb (hunk 7 above)
Comment (andreyruby) [reply HW02/andrey_bakanovsky/spec/array_refine_spec.rb]: I fixed it throught the tests.

**Pull request merged**